File: clients/c_sharp_client/client/core/baseStrategy.cs
using System;
using Api;
using System.Collections.Generic;
namespace client
{
	public abstract class baseStrategy
	{
		private DebugClass debug;
	
		public DebugClass Debug
		{
			set { debug = value; }
		}

		public baseStrategy()
		{
		}
			
		public abstract void onTick(List<Passenger> myPassengers, List<Elevator> myElevators, List<Passenger> enemyPassengers, List<Elevator> enemyElevators);

		public void log(Object obj)
		{
			debug.log (obj);
		}
	}
}

File: clients/go_client/client/src/client/core/API/BaseStrategy.go
package API

type BaseStrategy struct {
	mDebug *Debug
}

type IBaseStrategy interface {
	Init()
	Debug() *Debug
	SetDebug(debug *Debug)
	Log(object interface{})
	OnTick(myPassengers []*Passenger, myElevators []*Elevator,
		enemyPassengers []*Passenger, enemyElevators []*Elevator) error
}

func (s *BaseStrategy) Init() {

}

func (s *BaseStrategy) Debug() *Debug {
	return s.mDebug
}

func (s *BaseStrategy) SetDebug(debug *Debug) {
	s.mDebug = debug
}

func (s *BaseStrategy) Log(object interface{}) {
	s.mDebug.Log(object)
}

func (s *BaseStrategy) OnTick(myPassengers []*Passenger, myElevators []*Elevator,
	enemyPassengers []*Passenger, enemyElevators []*Elevator) error {

	return nil
}
